### DISCUSSION TITLE
What does `doCheck = true` do exactly

### DISCUSSION THREAD

**Discussion by dpc** [ipetkov/crane#68] [Q&A] [answered]
Is it only about running unit-tests from the package tests?

When I look at logs of dependencies being building I see `Checking somepackage 1.2.3` and `Compiling somepackage 1.2.3` . 

When I do `cargo build` locally I only ever get to see `Compiling somepackage 1.2.3`.

**Comment by ipetkov** [comment, marked as answer]
`doCheck` is understood by `mkDerivation` which decides if tests should be run or not (not all packages want to run their tests). Also when cross compiling nix will assume the value is set to `false`, but otherwise will honor whatever it is set to
